### PR TITLE
add pagination to user e2e

### PR DESCRIPTION
**Pull request by dsesami**
##### SUMMARY

e2e pagination

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION

```
AWX 4.0.0-25-g4e01b1157
```

### CODE REVIEW

File: awx/ui/test/e2e/fixtures.js [hunk 1]
```diff
@@ -306,17 +306,28 @@ const getUpdatedProject = (namespace = session) => {
...
+    name = `${namespace}-job-template`,
+    updateProject = true
 ) => {
-    const promises = [
-        getInventory(namespace),
-        getAdminMachineCredential(namespace),
-        getUpdatedProject(namespace)
-    ];
+    let promises;
+    if (updateProject) {
```
Comment (elyezer) [inline awx/ui/test/e2e/fixtures.js:321]: I think if you did something like (kinda of pseudo-code):

```
promises  = [
    getInventory(namespace),
    getAdminMachineCredential(namespace),
]

if updateProject {
    promises.append(getUpdatedProject(namespace))
} else {
    promises.append(getProject(namespace))
}
```

Would make it easier to read, I had to go line by line to figure that only the last item is different.

**Review by elyezer — APPROVED** [review]
LGTM, left a non-blocker comment.

File: awx/ui/test/e2e/tests/test-pagination.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    before: (client, done) => {
+        const resources = [
+            getAdminMachineCredential(namespace),
+            getOrganization(namespace),
+            getProject(namespace),
+            getInventorySource(namespace),
+        ];
+
+        for (let i = 0; i < 25; i++) {
```
Comment (unlikelyzero) [inline awx/ui/test/e2e/tests/test-pagination.js]: woot

File: awx/ui/test/e2e/tests/test-pagination.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        Promise.all(resources)
+            .then(() => {
+                done();
+            });
+
+        client
+            .login()
+            .waitForAngular()
+            .resizeWindow(1200, 1000);
```
Comment (unlikelyzero) [inline awx/ui/test/e2e/tests/test-pagination.js:30]: boo

File: awx/ui/test/e2e/fixtures.js (hunk 1 above)
Comment (dsesami) [reply awx/ui/test/e2e/fixtures.js:321]: Great call, I can make this adjustment.

File: awx/ui/test/e2e/tests/test-pagination.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    },
+
+    'Test job template pagination operations': client => {
```
Comment (unlikelyzero) [inline awx/ui/test/e2e/tests/test-pagination.js]: We should call this out as pagination with filter. We should also have a general pagination without filter test

**Review by unlikelyzero — APPROVED** [review]
We should call this out as a filtered pagination test and add a basic one without filter/restricted view.

**Pull request merged**